modules/viz/src/vizimpl.cpp

```diff
@@ -213,8 +242,8 @@ void cv::viz::Viz3d::VizImpl::showWidget(const String &id, const Widget &widget,
         follower->SetCamera(renderer_->GetActiveCamera());
     }
 
-    renderer_->AddActor(WidgetAccessor::getProp(widget));
-    (*widget_actor_map_)[id] = WidgetAccessor::getProp(widget);
+    renderer_->AddActor(actor);
+    (*widget_actor_map_)[id] = actor;
```

Comment:
That is a bug introduced. Actor may be zero in case of 2D widget. Please run tests for Viz, please write test for newly added features.